datasets/imagenet-1k/imagenet-1k.py

```diff
@@ -41,141 +40,94 @@
-                f"{data_dir} does not exist. Make sure you insert a manual dir via "
-                "`datasets.load_dataset('imagenet2012', data_dir=...)` that points to the "
-                "tar file downloaded for ImageNet 2012. Manual download instructions: "
-                f"{self.manual_download_instructions}"
-            )
-        val_prep_script = dl_manager.download(self.config.val_prep_script)
-        images_path = os.path.join(data_dir, "ILSVRC", "Data", "CLS-LOC")
+        archive = dl_manager.download(_DATA_URL)
+        val_prep_script = dl_manager.download(_IN1K_VAL_PREP_SCRIPT)
 
-        self._prep_validation(images_path, val_prep_script)
+        val_fixes = self._prep_val(val_prep_script)
 
         return [
             datasets.SplitGenerator(
                 name=datasets.Split.TRAIN,
-                gen_kwargs={"files": dl_manager.iter_files(os.path.join(images_path, "train"))},
+                gen_kwargs={"files": dl_manager.iter_archive(archive), "fixes": None, "split": "train"},
             ),
             datasets.SplitGenerator(
                 name=datasets.Split.VALIDATION,
-                gen_kwargs={"files": dl_manager.iter_files(os.path.join(images_path, "val"))},
+                gen_kwargs={
+                    "files": dl_manager.iter_archive(archive),
+                    "fixes": val_fixes,
+                    "split": "val",
+                },
             ),
             datasets.SplitGenerator(
                 name=datasets.Split.TEST,
-                gen_kwargs={"files": dl_manager.iter_files(os.path.join(images_path, "test")), "no_labels": True},
+                gen_kwargs={"files": dl_manager.iter_archive(archive), "fixes": None, "split": "test"},
             ),
         ]
 
-    def _generate_examples(self, files, no_labels=False):
+    def _generate_examples(self, files, fixes, split):
         """Yields examples."""
         idx = 0
-        for file in files:
-            _, file_ext = os.path.splitext(file)
-            if file_ext.lower() == self.IMAGE_EXTENSION:
-                output = {"image": file}
-
-                if not no_labels:
-                    label = os.path.basename(os.path.dirname(file))
-                    output["label"] = IMAGENET2012_CLASSES[label]
-                else:
-                    output["label"] = -1
-
-                yield idx, output
-                idx += 1
+        for path, file in files:
+            if split in path:
+                file_root, file_ext = os.path.splitext(path)
+                if file_ext.lower() == self.IMAGE_EXTENSION:
+                    output = {"image": {"path": path, "bytes": file.read()}}
+
+                    if split == "train":
+                        label = os.path.basename(os.path.dirname(file_root))
+                        output["label"] = IMAGENET2012_CLASSES[label]
+                    elif split == "val":
+                        label = fixes.get(os.path.basename(file_root))
```

Comment:
When will this actually be None. This will actually `KeyError` if the key is not present. Instead, I suggest you do this:
```suggestion
                        label = fixes.get(os.path.basename(file_root), None)
```

Reply:
It already returns None by default ;) no need to specify it
```py
>>> {}.get(1) is None
True
```